scripts/packages/VSCodeServer/src/eval.jl

```diff
@@ -174,13 +175,30 @@ function repl_runcode_request(conn, params::ReplRunCodeRequestParams)::ReplRunCo
                     if show_code
                         REPL_PROMPT_STATE[] = REPLPromptStates.Error
                     end
-                    @static if isdefined(Base, :current_exceptions)
+                    errs = @static if isdefined(Base, :current_exceptions)
                         EvalErrorStack(Base.current_exceptions(current_task()))
                     elseif isdefined(Base, :catch_stack)
                         EvalErrorStack(Base.catch_stack())
                     else
                         EvalError(err, catch_backtrace())
                     end
+                    if CAN_SET_ERR[]
+                        try
+                            istrivial = @static if isdefined(Base, :istrivialerror)
+                                Base.istrivialerror(errs.stack)
+                            else
+                                true
+                            end
+                            @static if @isdefined setglobal!
+                                istrivial || setglobal!(Main, :err, errs)
+                            else
+                                istrivial || ccall(:jl_set_global, Cvoid, (Any, Any, Any), Main, :err, errs)
```

Comment:
This probably shouldn't set `err` to the VSCodeServer specific error types, right? What's the Base behaviour for this?

Reply:
It contains an `ExceptionStack`. So possibly it should be set to the contents of an EvalErrorStack?